### DISCUSSION TITLE
Undo into Selection ignores Selection

### DISCUSSION THREAD

**Discussion by comod** [JetBrains/ideavim#640] [Q&A]
Hi!

Often i paste something onto something what i have selected. often i do mistakes on this step. undo into the selection and try something different onto this selection doesnt work. the selection is visible, but ideavim does not care about. so either ideavim should cancel the selection on undo or what i more would like is to get back to the previously state before the action onto the selection.

Reproduce:
Start with:
`this that`

type (starting with cursor on `t` of `this`):
`yw ft ve p u p`

Produce:
`this thatthat`

Instead of:
`this that` or even better `this this`, but this depends on configuration how to deal with registers

**Comment by ljchen1** [comment]
From my testing, in IDEA 2023.1.x with IdeaVim 2.2.0, it reproduces the `this that`.
What's your IDE and IdeaVIM version? If you are using the latest version already, can you see this issue with empty `~/ideavimrc` (you could backup this file before testing).

**Reply by comod** [reply]
no still "this thatthat" with completely empty .ideavimrc!  2023.2.1 + 2.4.1

**Comment by citizenmatt** [comment]
Please vote for and track this issue: [VIM-696](https://youtrack.jetbrains.com/issue/VIM-696/vim-selection-issue-after-undo)
